process/smartContract/hooks/vmAccountsDB.go

```diff
@@ -265,3 +433,10 @@ func (vadb *VMAccountsDB) IsInterfaceNil() bool {
 	}
 	return false
 }
+
+// SetCurrentHeader sets current header to be used by smart contracts
+func (vadb *VMAccountsDB) SetCurrentHeader(hdr data.HeaderHandler) {
```

Comment:
SC should not be aware of a block or header. A block or header contains SC.

Reply:
needs data from current round, which exists in header, and will be later signed by consensus

Reply:
o...k....